TicketsDemo.EF/Repositories/CSVTrainRepository.cs

```diff
@@ -0,0 +1,126 @@
+            {
+                csv.Read();
+                csv.ReadHeader();
+                while (csv.Read())
+                {
+                    var train = new Train
+                    {
+                        Id = csv.GetField<int>("Id"),
+                        Number = csv.GetField<int>("Number"),
+                        StartLocation = csv.GetField("StartLocation"),
+                        EndLocation = csv.GetField("EndLocation")
+                    };
+                    trains.Add(train);
+                }
+            }
+            return trains;
+        }
+
+        public Data.Entities.Train GetTrainDetails(int id)
+        {
+            var train = new Train();
+            using (var reader = new StreamReader(_path + "trains.csv"))
+            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+            {
+                csv.Read();
+                csv.ReadHeader();
+                while (csv.Read())
+                {
+                    if (csv.GetField<int>("Id") == id)
+                    {
+                        train = new Train
+                        {
+                            Id = csv.GetField<int>("Id"),
+                            Number = csv.GetField<int>("Number"),
+                            StartLocation = csv.GetField("StartLocation"),
+                            EndLocation = csv.GetField("EndLocation"),
+                            Carriages = new List<Carriage>()
+                        };
+                    }
+                }
+            }
+
+            using (var reader = new StreamReader(_path + "carriages.csv"))
+            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+            {
+                csv.Read();
+                csv.ReadHeader();
+                while (csv.Read())
+                {
+                    if (csv.GetField<int>("TrainId") == train.Id)
+                    {
+                        var carriage = new Carriage
+                        {
+                            Id = csv.GetField<int>("Id"),
+                            Type = (CarriageType)csv.GetField<int>("Type"),
+                            DefaultPrice = csv.GetField<int>("DefaultPrice"),
+                            TrainId = csv.GetField<int>("TrainId"),
+                            Number = csv.GetField<int>("Number"),
+                            Places = new List<Place>()
+                        };
+                        train.Carriages.Add(carriage);
+                    }
+                }
+            }
+
+            using (var reader = new StreamReader(_path + "places.csv"))
+            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+            {
+                csv.Read();
+                csv.ReadHeader();
+                while (csv.Read())
+                {
+                    if (csv.GetField<int>("CarriageId") == train.Carriages.FirstOrDefault().Id)
```

Comment:
As far as I can see, places will be initialized only for first carriage. Whoops.